subsys/bluetooth/host/hci_core.c

```diff
@@ -765,6 +765,24 @@ static void update_pending_id(struct bt_keys *keys, void *data)
 }
 #endif
 
+static struct bt_conn *find_valid_conn(bt_addr_le_t *peer_addr)
+{
+	struct bt_conn *conn;
+	u8_t le_states[] = {BT_CONN_CONNECT, BT_CONN_CONNECT_DIR_ADV};
+
+	/*
+	 * Make lookup to check if there's a connection object in
+	 * CONNECT or DIR_ADV state associated with passed peer LE address.
+	 */
+	for (u32_t i = 0; i < ARRAY_SIZE(le_states); i++) {
```

Comment:
Maybe I was unclear, but I meant that using a for-loop (even if you have just one) seems overkill if you always have the same fixed two iterations. Doing two if-statements is (at least for me) more readable to understand what the code is doing since you don't then have to look to the beginning of the function to figure out what the contents of the le_states array is. Or do you think it's likely that this array will be extended in the future? Another issue: I don't think the Zephyr coding style allows mixing code and declarations, i.e. variable declarations (such as `i`) should be in the beginning of a code block.

Reply:
>>> I don't think the Zephyr coding style allows mixing code and declarations, i.e. variable declarations (such as i) should be in the beginning of a code block.

@jhedberg I don't think there is any rule for that. No document requires that or forbids this c99 feature. People just don't do it. If I am wrong please point to the document - I could not find any for Zephyr or for Linux kernel.
I am asking as mixing code and variables greatly improves readability and reduces maintenance costs.
Of course we could limit ourselves and do it but I don't see absolutely any reason for it. It was requited by old compilers that would not be able to cope with such mixing but we are well into 21st century now...